Content/Items/Jungle/Weapons.HauntedDaggerStaff.cs

```diff
@@ -0,0 +1,509 @@
+				}
+				else
+				{
+					if (Main.rand.NextBool())
+					{
+						Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(2f, 2f), ModContent.DustType<Dusts.GlowFastDecelerate>(),
+							Main.rand.NextVector2Circular(2f, 2f), 0, new Color(70, 200, 100), 0.45f);
+					}
+
+					rotationalVelocity = Vector2.Lerp(rotationalVelocity, Projectile.DirectionTo(Target.Center), 0.15f);
+					Projectile.rotation = rotationalVelocity.ToRotation() + MathHelper.PiOver2;
+
+					if (attackTimer == 45)
+					{
+						Projectile.velocity += Projectile.DirectionTo(Target.Center) * 22f;
+
+						Helpers.Helper.PlayPitched("Effects/HeavyWhooshShort", 0.65f, Main.rand.NextFloat(-0.1f, 0.1f), Projectile.Center);
+					}
+
+					Vector2 direction = Target.Center - Projectile.Center;
+					direction.Normalize();
+					direction *= 23.5f;
+					Projectile.velocity = Vector2.Lerp(Projectile.velocity, direction, 0.075f);
+
+					if (rotTimer > 0)
+						rotTimer = 0;
+				}
+
+				if (!Target.active || Target.Distance(Owner.Center) > 1000f)
+				{
+					TargetWhoAmI = -1;
+					AttackDelay = maxAttackDelay;
+					attackTimer = 0;
+				}
+			}
+		}
+
+		public override void OnHitNPC(NPC target, NPC.HitInfo hit, int damageDone)
+		{
+			if (!Embedded && target.life > 0)
+			{
+				EmbeddedTarget = target;
+				Projectile.ai[0] = 1f;
+				Projectile.friendly = false;
+				Projectile.tileCollide = false;
+				enemyOffset = Projectile.position - target.position;
+				enemyOffset -= Projectile.velocity;
+				Projectile.netUpdate = true;
+			}
+
+			Helpers.Helper.PlayPitched("Impacts/StabTiny", 1f, Main.rand.NextFloat(-0.1f, 0.1f), Projectile.Center);
+
+			for (int i = 0; i < 6; i++)
+			{
+				Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(2f, 2f), ModContent.DustType<Dusts.GlowFastDecelerate>(),
+					-Projectile.velocity.RotatedByRandom(0.5f) * Main.rand.NextFloat(0.5f), 0, new Color(70, 200, 100), 0.5f);
+
+				if (Helpers.Helper.IsFleshy(target))
+				{
+					Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(3f, 3f), DustID.Blood,
+						-Projectile.velocity.RotatedByRandom(0.75f) * Main.rand.NextFloat(0.25f), Main.rand.Next(100), default, 2.5f).noGravity = true;
+
+					Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(3f, 3f), ModContent.DustType<Dusts.GraveBlood>(),
+						-Projectile.velocity.RotatedByRandom(0.75f) * Main.rand.NextFloat(0.25f), Main.rand.Next(100), default, 1.5f);
+				}
+				else
+				{
+					Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(2f, 2f), ModContent.DustType<Dusts.GlowFastDecelerate>(),
+					-Projectile.velocity.RotatedByRandom(0.5f) * Main.rand.NextFloat(0.5f), 0, new Color(150, 80, 30), 0.5f);
+
+				}
+			}
+
+			CameraSystem.shake += 2;
```

Comment:
outside the scope of this PR but we should really make screenshake scale off of where it actually hits so you aren't being shaken by someone elses minions across the map